src/averageAges.js

```diff
@@ -55,10 +79,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filterMothers = people
+    .filter(person =>
+      people.find(otherPerson => otherPerson.mother === person.name));
+
+  const filterChildren = people
+    .filter(person =>
+      people.find(otherPerson => otherPerson.name === person.mother));
+
+  const filterChildrenSons = people
+    .filter(person =>
+      people.find(otherPerson => otherPerson.name === person.mother)
+      && person.sex === 'm');
```

Comment:
try to unite three filter iterations in one. use ternary inside filter callback. after that will be no need make ternary for reduce method.
better to use some method instead find as we need just to know exist or not son in mother.